### PR TITLE
BICAWS-2912: Lock case when defendant name is clicked

### PR DESCRIPTION
**Pull request by ian-antking**

### CODE REVIEW

File: src/features/CourtCaseList/CourtCaseListEntry/CaseDetailsRow/CaseDetailsRow.tsx [hunk 1]
```diff
@@ -96,7 +110,11 @@ export const CaseDetailsRow = ({
           </ConditionalRender>
         </Table.Cell>
         <Table.Cell className={caseDetailsCellClass}>
-          <Link href={caseDetailsPath(errorId)} id={`Case details for ${defendantName}`}>
+          <Link
+            onClick={async (event) => lockAndOrViewCase(event)}
+            href="" // emply href enables expected mouseover behaviour
```
Comment (emadkaramad) [inline src/features/CourtCaseList/CourtCaseListEntry/CaseDetailsRow/CaseDetailsRow.tsx:115]: I wonder if we can use `<ActionLink>` component here. It does generate a link-like button without the need to set the href attribute.

**Review by emadkaramad — DISMISSED** [review]

File: src/features/CourtCaseList/CourtCaseListEntry/CaseDetailsRow/CaseDetailsRow.tsx (hunk 1 above)
Comment (ian-antking) [reply src/features/CourtCaseList/CourtCaseListEntry/CaseDetailsRow/CaseDetailsRow.tsx:115]: :eyes:

Comment (ian-antking) [reply src/features/CourtCaseList/CourtCaseListEntry/CaseDetailsRow/CaseDetailsRow.tsx:115]: Have swapped it out. Basically the same implementation, but without the need for an empty href and preventing default behaviour.

**Review by emadkaramad — APPROVED** [review]

**Review by richardcane — APPROVED** [review]

**Comment by ian-antking** [comment]
Reverted from using an ActionLink because it changed the element type from `a` to `button`, thus breaking the tests. Using a button for navigation also affects accessibility.

**Pull request merged**